### ISSUE TITLE
Line Numbers During Action Code (Again)

### ISSUE THREAD

**Issue by HiDefender** [lalrpop/lalrpop#495]
I know this issue has been raised in [323 ](https://github.com/lalrpop/lalrpop/issues/323)  and [138](https://github.com/lalrpop/lalrpop/issues/138), but I think I might have a way around it that isn't too clumsy.

@Marwes comments here about the approach that Gluon takes. Couldn't the `Lines` struct [from Gluon](https://github.com/gluon-lang/gluon/blob/f8326d21a14b5f21d203e9c43fa5bb7f0688a74c/base/src/source.rs) be generated prior to lexing. It could then be passed into the parser.

Since `Lines` is available to the parser, the line and column number could be found with `<start:@L> TOKEN <end:@R>`.

Am I on to something or not?

**Comment by Marwes** [comment]
Yes, that would certainly work if you need line/column during parsing.

**Comment by HiDefender** [comment]
I'm not extremely familiar with Lalrpop, but can put in some time towards this. It would be nice if it were a built-in macro.

Perhaps at the top of the `parser.lalrpop` file the user inserts a statement that generates the `Lines` struct.

Then some
```
A <b:B@> <c:C@> D => {
        let (b_value, b_left_line, b_left_col, b_right_line, b_right_col) = b;
        let (c_value, c_left_line, c_left_col, c_right_line, c_right_col) = c;
        assert_eq!(b_right_col + 1, c_left_col);
    }
```

I'm not sure what behavior would look like if `B` and `C` were not literals.
